### PR TITLE
Improve import functionality. Allow modal uploading. 

### PR DESCRIPTION
**Pull request by srdecny**
Rozsiril jsem moznost importu o nahrani jednotlivych souboru (rozpocet, faktury, akce) zvlast a moznost prepsani/pridani ke stavajicim zaznamum. Pridal jsem i logovani procesu importu, aby uzivatel, co importuje, mel moznost snadno prijit na to, kde v datech ma chybu. Importy take funguji stylem "vse nebo nic", tzn. se bud naimportuje cely soubor (pokud je OK), nebo se naopak nenaimportuje nic, pokud je v souboru chyba.

### CODE REVIEW

File: client/src/app/services/import.service.ts [hunk 1]
```diff
@@ -13,8 +13,34 @@ export class ImportService {
 
   constructor(private http: HttpClient) { }
 
-  async importAccounting(profileId: Profile["id"], formData: FormData) {
-    return this.http.post(this.root + "/profiles/" + profileId + "/accounting", formData, { responseType: 'text' }).toPromise();
+  async importAccounting(profileId: Profile["id"], formData: FormData, isAppend?: boolean) {
```
Comment (vojtan) [inline client/src/app/services/import.service.ts:16]: Too much repeating code. Maybe a common method import that all the other methods would call?

File: client/src/app/views/admin/views/admin-profile/admin-profile-logs/admin-profile-logs.component.html [hunk 2]
```diff
@@ -1,26 +1,35 @@
-<table class="table">
+<table class="table" style="table-layout: fixed; width: 100%;">
```
Comment (vojtan) [inline client/src/app/views/admin/views/admin-profile/admin-profile-logs/admin-profile-logs.component.html:1]: Avoid styling through style attribute extract to a class

File: client/src/app/views/admin/components/data-upload-modal/data-upload-modal.component.ts [hunk 3]
```diff
@@ -25,33 +30,62 @@ export class DataUploadModalComponent implements OnInit {
...
+            tasks.push(this.importService.importData(this.profileId, data, v[mode] == "append"))
+            break
+        }
+      } 
+    }
 
+    await Promise.all(tasks).then(_ => {
+      this.toastService.toast("Data nahrána pro import.", "notice")
+    }).catch(e => {
+      this.toastService.toast("Neočekáváná chyba při odesílání dat. Prosím, kontaktujte administrátora.", "notice")
```
Comment (vojtan) [inline client/src/app/views/admin/components/data-upload-modal/data-upload-modal.component.ts:85]: All the toasts in this component are of the type "notice". Consider using "error" type where appropriate

File: server/src/worker/import/parser.ts [hunk 4]
```diff
@@ -18,168 +19,209 @@ const headerNames = {
...
-      eventsFile.pipe(parser).pipe(reader);
-    });
+  static createCsvTransformer(file: string, options: Importer.Options): Transform {
+    switch (file) {
+      case "paymentsFile":
+      case "dataFile":
+      case "accountingFile":
+        return this.createDataTransformer(options)
+      case "eventsFile":
+        return this.createEventsReader(options)
```
Comment (vojtan) [inline server/src/worker/import/parser.ts:82]: Naming unclear is it a reader or a transformer?

File: server/src/routers/import/accounting.ts [hunk 5]
```diff
@@ -31,17 +31,143 @@ const importAccountingSchema = {
...
+		.first();
+
+	if (!year) {
+		const yearInsert = await db<YearRecord>("app.years").insert({ profileId: Number(req.params.profile), year: req.body.year }, ["profileId", "year"]).then()
+		year = yearInsert ? yearInsert[0] : null;
+		if (!year) return res.status(500).send("Failed to create new accounting year in database.");
+	}
+
+    // add import task to database queue (worker checks the table)
+	var importData: Partial<ImportRecord> = {
```
Comment (vojtan) [inline server/src/routers/import/accounting.ts:127]: If I import several files at once. It seems that it creates one job per one file type? Is that the correct behavior. How does it handle the rollback in this scenario?

Comment (srdecny) [reply server/src/routers/import/accounting.ts:127]: Yes, if you upload (e.g) three files in the upload form, three separate jobs are created. Each job has it's own DB transaction, so the rollbacks are handled separately for each file. The exception is the upload via .zip, which runs in a single job, no matter the number of files in the .zip.

Comment (vojtan) [reply server/src/routers/import/accounting.ts:127]: maybe update the documentation.md to clarify this behavior to the users.

File: client/src/app/services/import.service.ts (hunk 1 above)
Comment (srdecny) [reply client/src/app/services/import.service.ts]: Agreed; method created.

File: client/src/app/views/admin/views/admin-profile/admin-profile-logs/admin-profile-logs.component.html (hunk 2 above)
Comment (srdecny) [reply client/src/app/views/admin/views/admin-profile/admin-profile-logs/admin-profile-logs.component.html]: Removed; seems like it isn't needed anyways.

File: server/src/worker/import/parser.ts (hunk 4 above)
Comment (srdecny) [reply server/src/worker/import/parser.ts]: Nice catch; fixed.

**Review by vojtan — APPROVED** [review]

**Pull request merged**